austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,99 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/mob/living/simple_animal/hostile/mining_drone/Initialize()
+	. = ..()
+	weather_immunities += "ash" // no damage from ash storms
+	stored_gun.overheat_time = 10
+	var/datum/action/innate/minedrone/marker_beacon/beacon_action = new()
+	beacon_action.Grant(src)
+/datum/action/innate/minedrone/marker_beacon
+	name = "Drop Marker"
+	button_icon_state = "mech_zoom_off"
+
+/datum/action/innate/minedrone/marker_beacon/Activate()
+	var/obj/structure/marker_beacon/M = new(get_turf(owner))
+	to_chat(owner, "<span class='notice'>You place a [M].</span>")
```

Comment:
This should have a max amount or at the very least a synthesis cooldown

Reply:
I've applied a cap on marker beacons.  Both per minebot and globally.